src/error.rs

```diff
@@ -102,11 +119,10 @@ impl RenderError {
     }
 
     pub fn strict_error(path: Option<&String>) -> RenderError {
-        let msg = match path {
-            Some(path) => format!("Variable {:?} not found in strict mode.", path),
-            None => "Value is missing in strict mode".to_owned(),
-        };
-        RenderError::new(msg)
+        match path {
+            Some(path) => RenderErrorReason::MissingVariable(path.to_owned()).into(),
+            None => RenderError::new("Value is missing in strict mode"),
```

Comment:
Because both cases are `MissingVariable`. How about change this block to 

```rust
RenderErrorReason::MissingVariable(path.map(|p| p.to_owned())).into()
```

And make `MissingVariable(String)` hold `MissingVariable(Option<String>)`